### ISSUE TITLE
your sample code is not processing the very last pixel in your png example.

### ISSUE THREAD

**Issue by bvdeenen** [pedropark99/zig-book#97]
Hi Pedro
I was trying the Zig array out of bounds check while having built your [png example](https://pedropark99.github.io/zig-book/Chapters/13-image-filter.html#applying-the-image-filter) and just added `+2` to the `buffer[index + ..]` statements.
To my surprise the array bounds checker didn't trigger.
A bit of searching led me the conclusion your `while` loop limit is wrong. It should just be `index<len`. The limit is checked before the loop body is executed, the increment `(index+=4)` after the loop body has been executed.

Thanks for the great work!

**Comment by pedropark99** [comment]
Thank you @bvdeenen! I will take a look at it

**Comment by bvdeenen** [comment]
> Thank you @bvdeenen! I will take a look at it

I just checked to make sure, and I'm right:
```
    var index: u32 = 0;
    const len = 12;
    while (index < len - 4) : (index += 4) {
        try stdout.print("index={},  ", .{index});
    }
```
This prints 0 and 4, whereas the last one should be `8`.

**Comment by pedropark99** [comment]
Solved by #98

**Comment by pedropark99** [comment]
Thanks, fixed.
